class.convert-classic-to-gb-blocks.php

```diff
@@ -300,43 +444,70 @@ public static function ccetgb_convert_classic_post_to_gb_blocks($content = '', $
                 }
 
                 if (strpos($content, '<iframe') !== false) {
-                    $content = str_replace('<iframe', '<!-- wp:html --> <iframe', $content);
-                    $content = str_replace('</iframe>', '</iframe><!-- /wp:html -->', $content);
+                    $content = str_replace('<p>','',$content);
+                    $content = str_replace('</p>','',$content); 
+                    $return_content = self::ccetgb_convert_iframe($content);
+                    $start = '<iframe';
+                    $end = '</iframe>';
+                    $pattern = sprintf('/%s(.+?)%s/ims',preg_quote($start, '/'), preg_quote($end, '/'));
+                    if (preg_match($pattern, $content, $matches)) {
+                        list(, $match) = $matches;
+                       
+                    }
+                    $content = str_replace('<iframe'.$match.'</iframe>',$return_content,$content);
+                    $content = str_replace('<p>','',$content);
+                    $content = str_replace('</p>','',$content); 
+                    $content = str_replace('<!-- wp:paragraph --> <p>','',$content);
+                    $content = str_replace('</p><!-- /wp:paragraph -->','',$content);
                 }
 
                 if (strpos($content, '<p') !== false) {
-                    $content = str_replace('<p', '<!-- wp:paragraph --> <p', $content);
-                    $content = str_replace('</p>', '</p><!-- /wp:paragraph -->', $content);
+                    $content = preg_replace("/<p>/i",'<!-- wp:paragraph --> <p>', $content);
```

Comment:
Make sure class attribute must be add dynamically throughout all element. Please reduce 3 line of code to two line code for replacing element. try following regex.
`/<p(|( [a-z][a-z0-9]*)[^>]*?(\/?))>/i`